addons/mass_mailing/static/src/js/mailing_m2o_filter.js

```diff
@@ -63,8 +63,12 @@ export class FieldMany2OneMailingFilter extends Many2OneField {
         const recordDomain = new Domain(this.props.record.data[this.props.domain_field] || []).toString();
         const filterDomain = new Domain(this.props.record.data.mailing_filter_domain || []).toString();
         el.classList.toggle('d-none', recordDomain === '[]');
+        let inputElement = document.querySelector(`input#${this.props.model_field}`);
+        if(inputElement && inputElement.value){
```

Comment:
(comment wasn't submited)
```suggestion
        if (inputElement && inputElement.value) {
```